c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1717,6 +2078,17 @@ class DeviceCachingAllocator {
     }
   }
 
+  void addPeerAccess(int dev_to_access) {
+    devices_with_peer_access_.push_back(dev_to_access);
```

Comment:
should we skip this function if `dev_to_access` is already in `devices_with_peer_access_`